habitat/core/env.py

```diff
@@ -423,3 +424,238 @@ def __enter__(self):
 
     def __exit__(self, exc_type, exc_val, exc_tb):
         self.close()
+
+
+class CLEnv(Env):
+    r"""Environment base class extension designed to handle a stream-line of tasks.
+
+    The major :ref:`task` component is hereby intended as current task with which
+    the agent is currently interacting.
+    All tasks handled by the CLEnv are available through the :ref:`tasks` object
+    property. 
+    
+    The environment abstracts away the serving of multiple tasks under
+    the scheduling preferences expressed in the configuration.
+    """
+    _tasks: List[EmbodiedTask]
+    DEFAULT_EPS_CHANGE = 10
+
+    def __init__(
+        self, config: Config, dataset: Optional[Dataset] = None
+    ) -> None:
+        """ Constructor. Mimics the :ref:`Env` constructor.
+
+        Args:
+            :param config: config for the environment. Should contain id for
+            simulator and ``task_name`` for each task, which are passed into ``make_sim`` and
+            ``make_task``.
+            :param dataset: reference to dataset used for first task instance level
+            information. Can be defined as :py:`None` in which case
+            dataset will be built using ``make_dataset`` and ``config``.
+        """
+        # we let superclass instantiate first task as to not break/change habitat code
+        super().__init__(config, dataset=dataset)
+        # instatiate other tasks
+        self._tasks = [self._task]
+        self._curr_task_idx = 0
+        if len(config.TASKS) > 1:
```

Comment:
Unnecessary if check, as long as there is one TASK in the config.TASKS it should work. If there is only 1, the loop will not execute because config.TASKS[1:] will be empty.

Reply:
Thanks a lot, I'm going to fix that.